lib/gtfsplus/actions/gtfsplus.js

```diff
@@ -173,107 +175,104 @@ export function uploadGtfsPlusFeed (feedVersionId: string, file: File) {
-      .then(result => {
-        console.log(result)
-        return dispatch(uploadedGtfsPlusFeed())
-      })
+      .then(result => dispatch(uploadedGtfsPlusFeed()))
   }
 }
 
+type GTFSPlusEntity = {
+  rowIndex: number,
+  origRowIndex: number,
+  stop_id?: string,
+  route_id?: string,
+  [string]: any
+}
+
 export function loadGtfsEntities (
   tableId: string,
-  rows: Array<Route | Stop>,
-  feedSource: Feed,
-  feedVersionId: string
+  rows: Array<GTFSPlusEntity>,
+  feedVersion: FeedVersion,
+  validation: GtfsPlusValidation
 ) {
   return function (dispatch: dispatchFn, getState: getStateFn) {
-    // lookup table for mapping tableId:fieldName keys to inputType values
-    const typeLookup = {}
-    const getDataType = function (tableId, fieldName) {
-      const lookupKey = tableId + ':' + fieldName
-      if (lookupKey in typeLookup) return typeLookup[lookupKey]
-      const specTable = getGtfsPlusSpec().find(t => t.id === tableId)
-      if (!specTable) {
-        console.warn(`unable to find spec table for tableId: ${tableId}`)
-        return null
-      }
-      const fieldInfo = specTable.fields.find(f => f.name === fieldName)
-      if (!fieldInfo) return null
-      typeLookup[lookupKey] = fieldInfo.inputType
-      return fieldInfo.inputType
+    const tableIssues = validation[tableId]
+    // Filter out any rows that have missing identifiers. We don't want to include
+    // these in the subsequent GraphQL query.
+    const filteredRows = tableIssues
+      ? rows.filter(row => {
+          if (row.stop_id) {
+            const missingIdIssue = tableIssues.find(issue =>
+              issue.fieldName === 'stop_id' && issue.rowIndex === row.origRowIndex)
+            if (missingIdIssue) return null
+          }
+          if (row.route_id) {
+            const missingIdIssue = tableIssues.find(issue =>
+              issue.fieldName === 'route_id' && issue.rowIndex === row.origRowIndex)
+            if (missingIdIssue) return null
+          }
+          return row
+        })
+      : rows
+    const {namespace} = feedVersion
+    const {routesToLoad, stopsToLoad} = lookupMissingEntities(tableId, filteredRows, getState().gtfsplus.gtfsEntityLookup)
+    if (routesToLoad.length === 0 && stopsToLoad.length === 0) {
+      return
     }
+    dispatch(fetchGraphQL({
+      query: stopsAndRoutes(namespace, routesToLoad.length > 0 ? routesToLoad : null, stopsToLoad.length > 0 ? stopsToLoad : null),
+      variables: {namespace, routeId: routesToLoad, stopId: stopsToLoad}
+    }))
+      .then(results => dispatch(receiveGtfsEntities(results)))
+  }
+}
 
-    // determine which routes, stops, etc. aren't currently in the gtfsEntityLookup table and need to be loaded from the API
-    const routesToLoad = []
-    const stopsToLoad = []
-
-    const currentLookup = getState().gtfsplus.gtfsEntityLookup
+function lookupMissingEntities (tableId: string, rows: Array<GTFSPlusEntity>, currentLookup: Object) {
```

Comment:
This is a great candidate for making a unit test for.

Reply:
I have simplified this function quite a bit and added comments. Hopefully, that should be sufficient for improving maintainability at the moment.